source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,144 @@
+.. _fundamentals-kotlin-serialization:
+
+====================
+Kotlin Serialization
+====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+The Kotlin driver supports the ``kotlinx.serialization`` library for serializing 
+and deserializing Kotlin objects. 
+
+The driver provides an efficient ``Bson`` serializer that can can be used with 
+``@Serializable`` classes to handle the serialization of Kotlin objects to BSON data. 
+The ``bson-kotlinx`` library also supports :ref:`custom codecs <kotlin-custom-codec>`
+with configurations such as encoding defaults, encoding nulls, and defining class 
+discriminators.
+
+.. note::
+
+    To use the ``Codec`` interface to specify custom encoding and decoding of 
+    Kotlin objects to BSON data, see the :ref:`Codecs <fundamentals-codecs>` documentation.
+
+Although you can use the Kotlin driver with the Kotlin Serialization's ``Json`` 
+library, the ``Json`` serializer does *not* directly support BSON value types such
+as ``ObjectId``. You must provide a custom serializer that can handle the 
+conversion between BSON and JSON. 
+
+Add Kotlin Serialization to Your Project
+----------------------------------------
+
+The Kotlin driver serialization support depends on the the official Kotlin 
+Serialization library. You must add `Kotlin Serialization <https://github.com/Kotlin/kotlinx.serialization>`__ 
+to your project:
+
+.. tabs::
+
+   .. tab::
+      :tabid: Gradle
+
+      If you are using `Gradle <https://gradle.org/>`__, add the following to your
+      ``build.gradle.kts`` dependencies list:
+
+      .. code-block:: kotlin
+        :caption: build.gradle.kts
+
+        implementation("org.jetbrains.kotlinx:kotlinx-serialization-core:{+serializationVersion+}")
+
+   .. tab::
+      :tabid: Maven
+
+      If you are using `Maven <https://maven.apache.org/>`__, add the following to
+      your ``pom.xml`` dependencies list:
+
+      .. code-block:: kotlin
+        :caption: pom.xml
+   
+        <dependency>
+            <groupId>org.jetbrains.kotlinx</groupId>
+            <artifactId>kotlinx-serialization-core</artifactId>
+            <version>{+serializationVersion+}</version>
+        </dependency>
+
+Annotate Data Classes
+---------------------
+
+Once added, you can annotate your data classes with ``@Serializable`` annotation 
+that comes from the Kotlin Serialization framework.
+
+.. note:: 
+
+    You cannot use :ref:`annotations <fundamentals-data-class-annotations>` 
+    from the ``org.bson.codecs.pojo.annotations`` package on ``@Serializable`` data classes.
+
+.. literalinclude:: /examples/generated/KotlinXSerializationTest.snippet.basic-serialization.kt
+    :language: kotlin
+
+.. _kotlin-custom-codec:
+
+Create a Custom Codec
```

Comment:
Suggestion change title to relate to customizing what is stored.

Reply:
Renamed -- let me know if it still needs tweaking